logfile.go

```diff
@@ -20,26 +22,44 @@ func lineWorker(die dieCh, r *bufio.Reader, cfg logplexc.Config, sr *serveRecord
 		log.Fatalf("could not create logging client: %v", err)
 	}
 
-	for {
-		select {
-		case <-die:
-			return
-		default:
-			break
-		}
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer watcher.Close()
 
-		l, _, err := r.ReadLine()
-		l = prefix.ReplaceAll(l, []byte(""))
-		if len(l) > 0 {
-			target.BufferMessage(134, time.Now(), "redis",
-				sr.Name, l)
-		}
+	r := bufio.NewReader(f)
 
-		if err != nil {
-			if err == io.EOF {
-				continue
+	go func() {
+		for {
+			select {
+			case event := <-watcher.Events:
+				if event.Op&fsnotify.Write == fsnotify.Write {
```

Comment:
I understand the principle of operation on why this helps with performance, yet I have a high level question: is *everyone* writing programs that takes redis logs via this FIFO doing this kind of non-portable fs watching in order to get decent performance, or is there some obscure unixism other implementations are using to avoid it?